Add "Gradient Descent" to AI topic (#2195)
Codecademy/docs: content/ai/concepts/search-algorithms/terms/gradient-descent/gradient-descent.md

---
Title: 'Gradient Descent'
Description: 'Gradient descent is an optimization search algorithm that is widely used in machine learning to train neural networks and other models.'
Subjects:
  - 'Data Science'
  - 'AI'
Tags:

>   - 'Search Algorithms'

Reviewer: ```suggestion
  - 'Search'
  - 'Algorithms'
```

  - 'AI'
CatalogContent:
  - 'learn-python-3'
  - 'paths/machine-learning-ai-engineering-foundations'
---

**Gradient descent** is an optimization search [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that is widely used in [machine learning](https://www.codecademy.com/resources/docs/ai/machine-learning) to train neural networks and other models. It is a fundamental optimization algorithm that is used in a many fields of [artificial intelligence](https://www.codecademy.com/resources/docs/ai), such as natural language processing, computer vision, and robotics.

## How Does Gradient Descent Work?

Gradient descent works by minimizing the cost (or loss) function of a model. Cost functions measure the performance of machine learning models by quantifying the difference between the predicted and actual outputs. A low cost function indicates that a model is performing well on the training data because its predicted output values are close to the actual output values.

> Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values for the model to achieve good performance.

Reviewer: ```suggestion
Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.
```

Subjects:
  - 'Data Science'
  - 'AI'
Tags:
  - 'Search'
  - 'Algorithms'
  - 'AI'
CatalogContent:
  - 'learn-python-3'
  - 'paths/machine-learning-ai-engineering-foundations'
---

**Gradient descent** is an optimization search [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that is widely used in [machine learning](https://www.codecademy.com/resources/docs/ai/machine-learning) to train neural networks and other models. It is a fundamental optimization algorithm that is used in a many fields of [artificial intelligence](https://www.codecademy.com/resources/docs/ai), such as natural language processing, computer vision, and robotics.

## How Does Gradient Descent Work?

Gradient descent works by minimizing the cost (or loss) function of a model. Cost functions measure the performance of machine learning models by quantifying the difference between the predicted and actual outputs. A low cost function indicates that a model is performing well on the training data because its predicted output values are close to the actual output values.

> Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood”. The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.

Reviewer 2: ```suggestion
Gradient descent is an iterative optimization algorithm, which means that it iteratively, or repeatedly, updates the parameters of a model until a specific outcome is achieved. Parameters can be thought of as the “settings” for a machine learning model. The purpose of gradient descent is to iterate until the local or global minimum is found in a given solution space or “local neighborhood." The “gradient” refers to the rate of change, or slope, of the cost function. Thus, a gradient descent algorithm iteratively moves down (descends) the gradient (slope of the cost function) to find the minimum point, which represents the optimal values (the best performance) for the model in the given context.
```

Reviewer 2: very minor suggestion
